cli/src/semgrep/target_manager.py

```diff
@@ -176,6 +176,20 @@ def unsupported_lang_paths(self) -> FrozenSet[Path]:
             else self.target_manager.get_all_files()
         )
 
+    def list_skipped_paths_with_reason(self) -> List[Tuple[Path, str]]:
+        res: List[Tuple[Path, str]] = []
+        # The strings used to describe the reason are those defined
+        # in semgrep_output_v1.atd for type 'skip_reason':
```

Comment:
then would be better to use the atd type there, and just call .to_json() on it later,
so the return type of this function would be List[Tuple[Path, out.SkipReason]]